orchestrator.go

```diff
@@ -380,20 +380,21 @@ func (s *SmartContract) queryItems(APIstub shim.ChaincodeStubInterface, args []s
 	objectType := args[0]
 	fmt.Printf("- start looking for elements of type %s\n", objectType)
 	resultsIterator, _ := APIstub.GetStateByRange(objectType+"_", objectType+"_z")
-	fmt.Println(resultsIterator)
-	results := make([]string, 0)
+	var payloadSlice [][]byte
 	for resultsIterator.HasNext() {
 		queryResponse, err := resultsIterator.Next()
 		if err != nil {
 			return shim.Error(err.Error())
 		}
-		results = append(results, queryResponse.String())
-	}
-	payload, err := json.Marshal(results)
-	if err != nil {
-		return shim.Error(err.Error())
+		payloadSlice = append(payloadSlice, queryResponse.GetValue())
 	}
 	fmt.Printf("- end looking for elements of type %s\n", objectType)
+
+	// Formatting in json
+	payload := append([]byte("["), bytes.Join(payloadSlice, []byte(","))...)
+	payload = append(payload, "]"...)
+
```

Comment:
Did we not talk about this way on mattermost @camillemarini ?
Looks unmaintenable to me. Putting `[` and `]` looks hacky.

Reply:
@GuillaumeCisco no, we didn't talk about this way on mattermost. It's a fix from Max-Pol to be able to parse output from the chaincode. Do you have an idea how to do it more properly?
Also @mgaltier is writting tests on the chaincode, and I think he is also looking at this

Reply:
And there is another issue btw: the "key" is not sent, only the "value".

Reply:
@Max-Pol  When I call queryItems in the docker cli, I get the keys:
```
peer chaincode query -n mycc -c '{"Args":["queryItems", "algo"]}' -C $CHANNEL_NAME
```
returns:
```
Query Result: ["namespace:\"mycc\" key:\"algo_0\" value:\"{\\\"docType\\\":\\\"algo\\\",\\\"problem\\\":\\\"problem_1\\\"}\" ","namespace:\"mycc\" key:\"algo_1\" value:\"{\\\"docType\\\":\\\"algo\\\",\\\"problem\\\":\\\"problem_1\\\"}\" "]
```

Reply:
I was thinking about https://mattermost.morpheo.co/morpheo/pl/c3ntdod5rfdsbju1sywi4a8ykr
Json stringify

Reply:
I changed it, as you can see, it's a "vrai bonheur" to parse json :)